ruleset/sca/amazon/cis_amazon_linux_2022.yml

```diff
@@ -0,0 +1,5084 @@
+    compliance:
+      - cis: ["1.1.1.3"]
+      - cis_csc_v8: ["4.8"]
+      - cis_csc_v7: ["9.2"]
+      - cmmc_v2.0: ["CM.L2-3.4.7", "CM.L2-3.4.8", "SC.L2-3.13.6"]
+      - pci_dss_3.2.1: ["1.1.6", "1.2.1", "2.2.2", "2.2.5"]
+      - pci_dss_4.0: ["1.2.5", "2.2.4", "6.4.1"]
+      - soc_2: ["CC6.3", "CC6.6"]
+    condition: all
+    rules:
+      - "c:modprobe -n -v udf -> r:install /bin/false|Module udf not found"
+      - "not c:lsmod -> r:udf"
+      - 'd:/etc/modprobe.d -> r:\.*.conf -> r:blacklist\sudf'
+
+  # 1.1.2.1 Ensure /tmp is a separate partition.
+  - id: 31003
+    title: "Ensure /tmp is a separate partition."
+    description: "The /tmp directory is a world-writable directory used for temporary storage by all users and some applications."
+    rationale: "Making /tmp its own file system allows an administrator to set the noexecoption on the mount, making /tmpuseless for an attacker to install executable code. It would also prevent an attacker from establishing a hardlink to a system setuidprogram and wait for it to be updated. Once the program was updated, the hardlink would be broken and the attacker would have his own copy of the program. If the program happened to have a security vulnerability, the attacker could continue to exploit the known flaw. This can be accomplished by either mounting tmpfs to /tmp, or creating a separate partition for /tmp."
+    impact: "Since the /tmp directory is intended to be world-writable, there is a risk of resource exhaustion if it is not bound to a separate partition. Running out of /tmp space is a problem regardless of what kind of filesystem lies under it, but in a configuration where /tmp is not a separate file system it will essentially have the whole disk available, as the default installation only creates a single / partition. On the other hand, a RAM-based /tmp (as with tmpfs) will almost certainly be much smaller, which can lead to applications filling up the filesystem much more easily. Another alternative is to create a dedicated partition for /tmp from a separate volume or disk. One of the downsides of a disk-based dedicated partition is that it will be slower than tmpfs which is RAM-based. /tmp utilizing tmpfs can be resized using the size={size} parameter in the relevant entry in /etc/fstab."
+    remediation: "First ensure that systemd is correctly configured to ensure that /tmp mount for your environment will be mounted at boot time. # systemctl unmask tmp.mount . For specific configuration requirements of the /tmp mount for your environment, modify /etc/stab. Example using tmpfs with specific mount options: # tmpfs /tmp / tmpfs defaults,rw,nosuid,nodev,noexec,relatime,size=2G 0 0 . Example using a volume or disk with specific mount options, The source location of the volumne or disk will vary depending on your environment: # <device> /tmp <fstype> defaults,nodev,nosuid,noexec 0 0"
```

Comment:
There are extra spaces before the end of the sentence dots.

Reply:
Solved